configuration/env_var_processors.rst

```diff
@@ -269,10 +277,13 @@ Symfony provides the following env var processors:
         .. code-block:: php
 
             // config/packages/framework.php
-            $container->setParameter('env(TRUSTED_HOSTS)', '["10.0.0.1", "10.0.0.2"]');
-            $container->loadFromExtension('framework', [
-                'trusted_hosts' => '%env(json:TRUSTED_HOSTS)%',
-            ]);
+            use Symfony\Component\DependencyInjection\ContainerBuilder;
+            use Symfony\Config\FrameworkConfig;
+
+            return static function (ContainerBuilder $container, FrameworkConfig $framework) {
+                $container->setParameter('env(TRUSTED_HOSTS)', '["10.0.0.1", "10.0.0.2"]');
+                $framework->trustedHosts('%env(json:TRUSTED_HOSTS)%');
```

Comment:
This does not work. `$framework->trustedHosts()` expects an array.

Reply:
@nicolas-grekas Should we make any changes for this scenario?

Reply:
Hmm. Should we remove type hints?
Alternatively, should we accept array|EnvConfigurator?

Reply:
Hm.. that means removing type hints for all "leaf" values and add them in the doc block. 

```diff
    /**
+    * @param bool|EnvConfigurator $value
     * @default false
     * @return $this
     */
-   public function enabled(bool $value): self
+   public function enabled($value): self
    {
        $this->enabled = $value;
    
        return $this;
    }
```

Reply:
Correct. I don't see a better way. We could generate union types on SF6.

Reply:
Same. 
I'll make a PR to Symfony then. 

> We could generate union types on SF6.

Yes =)

Reply:
That's the same issue with data converted by normalizers everywhere

Probably the same issue as https://github.com/symfony/symfony-docs/pull/15269#discussion_r618154902

Reply:
> We could generate union types on SF6

Thast mean, we should be able to detect the allowed types. That could be doable with `beforeNormalization()->ifString` but not for `beforeNormalization()->ifTrue`

Reply:
Hm.. Im not sure it is the same as the "lock issue". This is related to env vars. 

See https://github.com/symfony/symfony/pull/40903